api/cliconfig/validate_listener.go

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !fips_140_3
+
+package cliconfig
+
+func IsValidListener(tlsDisable bool) error {
```

Comment:
This change will require a complementary enterprise PR if we go ahead. The enterprise version only checks one parameter of the `internalshared/configutil` struct that it got passed previously (the listener config), so I thought it would be easiest to just extract that one value rather than trying to expose some of the listener config into our module API.